ignite/services/chain/chain.go

```diff
@@ -140,12 +142,17 @@ func New(path string, options ...Option) (*Chain, error) {
 		return nil, err
 	}
 
+	cacheStorage, err := cache.NewChainStorage(app.Name)
```

Comment:
You are correct that multiple chains with the same name would share the cache. I hadn't thought about them also running at the same time. That would create a locking problem. The bolt.Open would hang until unlocked I think (which wouldn't happen).

So the solution is to either make the name of the cache more unique (perhaps using the whole path somehow in the name, or hashed or something), or figure out a way to share the cache. Like I wrote above, the bolt.Open doesn't like having multiple processes (or even threads) open the DB at the same time, which is why I added that global variable to keep track of it.

I can of course look for other storage solutions, perhaps there is something that is better suited for multi-process access. Or we would have to create a lock + open and close the DB constantly to make sure it isn't locked for others.

As for having a shared cache, I think it might be fine, but I worry also that it might create some strange behavior down the line if someone expects the cache to be chain/project-unique. Can't really think of something right off the top of my head though, so :man_shrugging: 

Let me know what you want to do.

Reply:
If the key > value's we use in the cache is %100 deterministic (and right now it seems like that) for cross chains, I think it could be OK to use a single shared cache for all (e.g. go mod also works like that).

This way, we don't also need to worry about finding unique names for chains. All we need to do is to provide a safe concurrent access for the cache package and the database we use I think.

Reply:
It looks like this should be doable, but will require every opening of the database to be closed.

It does get a bit more verbose this way though, a lot of:
```go
cacheRootDir, err := chainconfig.ConfigDirPath()
if err != nil {
    return err
}
cacheStorage, err := cache.NewStorage(cacheRootDir)
if err != nil {
    return err
}

// Use cacheStorage

if err := cacheStorage.Close(); err != nil {
    return err
}
```